tests/test_config_consistency.py

```diff
@@ -71,3 +72,86 @@ def test_status_code_202(self):
         assert spans[0]["type"] == "web"
         assert spans[0]["meta"]["http.status_code"] == "202"
         assert spans[0]["error"] == 1
+
+
+@scenarios.default
+@features.tracing_configuration_consistency
+class Test_Config_HttpClientErrorStatuses_Default:
+    """ Verify behavior of http clients """
+
+    def setup_status_code_400(self):
+        self.r = weblog.get("/make_distant_call", params={"url": "http://weblog:7777/status?code=400"})
+
+    def test_status_code_400(self):
+        assert self.r.status_code == 200
+        content = json.loads(self.r.text)
+        assert content["status_code"] == 400
+
+        interfaces.library.assert_trace_exists(self.r)
+        spans = [s for _, _, s in interfaces.library.get_spans(request=self.r, full_trace=True)]
```

Comment:
It took me way to long to find the `full_trace` parameter. In my tests the `get_spans` method was consistently returning only the root span so I started writing my own helper 😵‍💫.